src/lib/mcp.ts

```
@@ -0,0 +1,316 @@
+    call: (toolName: string, args: any) => Promise<any>;
+    get isConnected(): boolean;
+    on: (event: string, listener: (...args: any[]) => void) => void;
+    off: (event: string, listener: (...args: any[]) => void) => void;
+}
+
+// MCP Observability events
+export interface MCPObservabilityEvents {
+    'mcp:connect:start': { transport: MCPTransportOptions };
+    'mcp:connect:complete': { duration: number };
+    'mcp:connect:error': { error: Error; duration: number };
+    'mcp:disconnect': void;
```

Biome (1.9.4)

[error] void is confusing outside a return type or a type parameter.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)